includes/incoming-events/class-user-deleted.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Newspack Hub User Deleted Incoming Event class
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network\Incoming_Events;
+
+use Newspack_Network\Debugger;
+use Newspack_Network\Utils\Users as User_Utils;
+
+/**
+ * Class to handle the user deletion.
+ */
+class User_Deleted extends Abstract_Incoming_Event {
+	/**
+	 * Processes the event
+	 *
+	 * @return void
+	 */
+	public function post_process_in_hub() {
+		$this->process_user_deleted();
+	}
+
+	/**
+	 * Process event in Node
+	 *
+	 * @return void
+	 */
+	public function process_in_node() {
+		$this->process_user_deleted();
+	}
+
+	/**
+	 * Process user deleted
+	 *
+	 * @return void
+	 */
+	public function process_user_deleted() {
+		$email = $this->get_email();
+		Debugger::log( 'Processing user deletion with email: ' . $email );
+		if ( ! $email ) {
+			return;
+		}
+		$user = get_user_by( 'email', $email );
+		if ( ! $user ) {
+			Debugger::log( sprintf( 'User to be deleted not found by email: %s, skipping.', $email ) );
+			return;
+		}
+
+		// Ensure this is a reader.
+		if ( ! \Newspack\Reader_Activation::is_user_reader( $user ) ) {
+			Debugger::log( sprintf( 'User %s is not a reader, skipping deletion.', $email ) );
+			return;
+		}
+
+		/** Make sure `wp_delete_user()` is available. */
+		require_once ABSPATH . 'wp-admin/includes/user.php';
+
+		// Don't broadcast this deletion on the network.
+		add_filter( 'newspack_network_process_user_deleted', '__return_false' );
```

Comment:
Just noting this results in conflicting messaging in the logs where the account is logged as skipped, but then is logged as deleted:

```
[25-Apr-2024 14:07:47 UTC] [3110] /Users/raz/Sites/newspack/network/includes/class-data-listeners.php:103 User deletion with email: raz+networksubscriber3@automattic.com was skipped due to filter use.
[25-Apr-2024 14:07:47 UTC] [3110] /Users/raz/Sites/newspack/network/includes/incoming-events/class-user-deleted.php:70 User raz+networksubscriber3@automattic.com deleted.
```

Since the final log indicates the deletion was successful, I think this is not a blocker.